drivers/regulator/regulator_shell.c

```diff
@@ -1,152 +1,404 @@
+
 	return 0;
 }
 
-static int cmd_set_mode(const struct shell *sh, size_t argc, char **argv)
+static int cmd_modeget(const struct shell *sh, size_t argc, char **argv)
 {
-	int mode, ret;
-	const struct device *reg_dev;
+	const struct device *dev;
+	regulator_mode_t mode;
+	int ret;
+
+	ARG_UNUSED(argc);
 
-	reg_dev = device_get_binding(argv[1]);
-	if (reg_dev == NULL) {
-		shell_error(sh, "regulator device %s not available", argv[1]);
+	dev = device_get_binding(argv[1]);
+	if (dev == NULL) {
+		shell_error(sh, "Regulator device %s not available", argv[1]);
 		return -ENODEV;
 	}
 
-	mode = strtol(argv[2], NULL, 10);
-	ret = regulator_set_mode(reg_dev, mode);
+	ret = regulator_get_mode(dev, &mode);
 	if (ret < 0) {
-		shell_error(sh, "failed to set mode, error %d", ret);
+		shell_error(sh, "Could not get mode (%d)", ret);
+		return ret;
 	}
-	return ret;
+
+	shell_print(sh, "Mode: %u", (unsigned int)mode);
+
+	return 0;
 }
 
-SHELL_STATIC_SUBCMD_SET_CREATE(regulator_set,
+static int cmd_errors(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	regulator_error_flags_t errors;
+	int ret;
+
+	ARG_UNUSED(argc);
+
+	dev = device_get_binding(argv[1]);
+	if (dev == NULL) {
+		shell_error(sh, "Regulator device %s not available", argv[1]);
+		return -ENODEV;
+	}
+
+	ret = regulator_get_error_flags(dev, &errors);
+	if (ret < 0) {
+		shell_error(sh, "Could not get error flags (%d)", ret);
+		return ret;
+	}
+
+	shell_print(sh, "Overvoltage:\t[%s]",
+		    ((errors & REGULATOR_ERROR_OVER_VOLTAGE) != 0U) ? "X"
+								    : " ");
+	shell_print(sh, "Overcurrent:\t[%s]",
+		    ((errors & REGULATOR_ERROR_OVER_CURRENT) != 0U) ? "X"
+								    : " ");
+	shell_print(sh, "Overtemp.:\t[%s]",
+		    ((errors & REGULATOR_ERROR_OVER_TEMP) != 0U) ? "X" : " ");
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	sub_regulator_cmds,
 	SHELL_CMD_ARG(enable, NULL,
-		"Enable regulator\n"
-		"Usage: enable <device>", cmd_reg_en, 2, 0),
+		      "Enable regulator\n"
+		      "Usage: enable <device>",
+		      cmd_enable, 2, 0),
 	SHELL_CMD_ARG(disable, NULL,
-		"Disable regulator\n"
-		"Usage: disable <device>", cmd_reg_dis, 2, 0),
-	SHELL_CMD_ARG(set_vol, NULL, "Set voltage (in mV)\n"
-		"Usage: set_vol <device> <low limit (uV)> <high limit (uV)>",
-		cmd_set_vol, 4, 0),
-	SHELL_CMD_ARG(set_current, NULL, "Set current limit( in mA)\n"
-		"Usage: set_current <device> <low limit (uA)> <high limit (uA)>",
-		cmd_set_ilim, 4, 0),
-	SHELL_CMD_ARG(set_mode, NULL, "Set mode of regulator\n"
-		"Usage: set_mode <device> <mode index>",
-		cmd_set_mode, 3, 0),
-	SHELL_SUBCMD_SET_END
-);
-
-SHELL_CMD_REGISTER(regulator, &regulator_set, "Regulator Management", NULL);
+		      "Disable regulator\n"
+		      "Usage: disable <device>",
+		      cmd_disable, 2, 0),
+	SHELL_CMD_ARG(vlist, NULL,
+		      "List all supported voltages\n"
+		      "Usage: vlist <device>",
+		      cmd_vlist, 2, 0),
+	SHELL_CMD_ARG(vset, NULL,
+		      "Set voltage\n"
```

Comment:
The parsing of volts and mV implemented in this command is very useful, but I think we should document it in the help string (maybe by clarifying the user should provide the unit with their input?)

Reply:
Sure, let me add a comment.